File: 806. 写字符串需要的行数/main.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {
        int lines = 1;
        int width = 0;
        for (auto & c : s) {
            int temp = widths[c - 'a']; // 当前字母所需要的长度
            width += temp;
            if (width > 100) { //如果加上该字母使得该行总长度>100，则将该字母放到下一行
                lines++;
                width = temp;
            }
        }
        return {lines, width};
    }
};

File: 806. 写字符串需要的行数/main.go
func numberOfLines(widths []int, s string) []int {
	const maxWidth = 100
	lines := 1
	width := 0
	for _, c := range s {
		temp := widths[c-'a']
		width += temp
		if width > maxWidth {
			lines++
			width = temp
		}
	}
	return []int{lines, width}
}